src/ansible/roles/ipa/tasks/main.yml

```diff
@@ -90,41 +101,77 @@
     ipa user-mod admin --password
   args:
     stdin: |
-      {{ service.ipa.password }}
-      {{ service.ipa.password }}
+      {{ ipa_password }}
+      {{ ipa_password }}
 
 - name: 'Check trust with other domains'
   shell: |
     kinit admin
     ipa trust-find
   args:
     stdin: |
-      {{ service.ipa.password }}
+      {{ ipa_password }}
   register: trust
   failed_when: False
 
-- name: 'Setup trust with {{ service.samba.domain }}'
-  shell: |
-    kinit admin
-    ipa trust-add {{ service.samba.domain | quote }} --admin Administrator --password
-  args:
-    stdin: |
-      {{ service.ipa.password }}
-      {{ service.samba.password }}
+- name: 'Setup trust with samba'
+  block:
+  - name: Set Samba facts (IPA)
+    set_fact:
+      samba_domain: "{{ '.'.join(hostvars[groups.samba.0].inventory_hostname.split('.')[1:]) }}"
+      samba_password: "{{ hostvars[groups.samba.0].ansible_password | default(service.ipa.password) }}"
+  - name: Debug Samba facts (IPA)
+    debug:
+      msg: 'Samba domain: "{{ samba_domain }}"'
+  - name: Run ipa trust-add
+    shell: |
+      kinit admin
+      ipa trust-add {{ samba_domain | quote }} --admin Administrator --password
+    args:
+      stdin: |
+        {{ ipa_password }}
+        {{ samba_password }}
+    when:
+    - 'samba_domain not in trust.stdout'
   when:
-  - 'service.samba.domain not in trust.stdout'
+  - '"samba" in groups and groups["samba"]'
   - join_samba
   - trust_ipa_samba
 
-- name: 'Setup trust with {{ service.ad.domain }}'
-  shell: |
-    kinit admin
-    ipa trust-add {{ service.ad.domain | quote }} --admin Administrator --password
-  args:
-    stdin: |
-      {{ service.ad.safe_password }}
-      {{ service.ad.safe_password }}
+- name: 'Setup trust with AD'
+  block:
+  - name: Set AD facts (IPA)
+    set_fact:
+      ad_domain: "{{ '.'.join(hostvars[groups.ad.0].inventory_hostname.split('.')[1:]) }}"
+      safe_password: "{{ service.ad.safe_password }}"
+      # TODO: range-type can be configurable ipa-ad-trust or ipa-ad-trust-posix
```

Comment:
Did you leave this TODO on purpose?

Reply:
Yes, I added it on purpose I wanted to discuss if it makes a sense to make it configurable from outside and discuss the default value as well.

Reply:
I don't have any expertise in this area. @pbrezina any opinion?

Reply:
In that case, let's resolve it and we can discuss it in the future